lib/run_part.c

```diff
@@ -51,47 +52,73 @@ int run_parts(const char *directory, const char *name, const char *action)
 	int n;
 	int execute_result = 0;
 
+	int dfd = open(directory, O_PATH | O_DIRECTORY | O_CLOEXEC);
+	if (dfd == -1) {
+		fprintf(shadow_logfd, "failed open directory %s: %s\n", directory, strerror(errno));
+		return (1);
+	}
+
+#ifdef HAVE_SCANDIRAT
+	scanlist = scandirat(dfd, ".", &namelist, 0, alphasort);
+#else
 	scanlist = scandir(directory, &namelist, 0, alphasort);
+#endif
```

Comment:
The oldest glibc still alive is glibc 2.19, used in Debian Jessie, which will be EOL in 2025.

I think we're good to assume that scandirat(3) is present in all systems that use shadow currently.

If anyone wants to port shadow to other systems, that's another story, but I'm not comfortable adding code that we know is dead code.

Reply:
Oh, my bad, musl doesn't have this.